homeassistant/components/onvif/camera.py

```diff
@@ -425,10 +405,11 @@ def __init__(self, hass, config):
     def setup_ptz(self):
         """Set up PTZ if available."""
         _LOGGER.debug("Setting up the ONVIF PTZ service")
-        if self._camera.get_service("ptz", create=False) is None:
```

Comment:
Same issue here, PTZ was not working for me because two functions to get the PTZ service were being called in quick succession.

After reducing this to a single function i can now use PTZ